src/sentry/consumers/__init__.py

```diff
@@ -396,12 +409,44 @@ def build_consumer_config(group_id: str):
             healthcheck_file_path, strategy_factory
         )
 
+    if enable_dlq:
+        try:
+            dlq_topic = consumer_definition["dlq_topic"]
+        except KeyError:
+            raise click.BadParameter(
+                f"Cannot enable DLQ for consumer: {consumer_name}, DLQ topic is not specified in consumer definition"
+            )
+        try:
+            cluster_setting = get_topic_definition(dlq_topic)["cluster"]
+        except ValueError:
+            raise click.BadParameter(
+                f"Invalid DLQ topic: {dlq_topic}, all DLQ topics must have topic definition configured in settings"
+            )
```

Comment:
Do we have no validation on consumer definitions before we get to this point? Can we add a validate method and put all validation in one place? Also, can we make sure all consumer definitions are validated in CI? It should not even be possible for anyone to check in a bad one to the codebase

Reply:
I don't think we do, all validation is done in `get_stream_processor` because there are multiple ways to call this function (i.e. `consumer.py`, `run.py`)

Reply:
Added a validation function and a test that runs every sentry consumer definition through the validation test